bundles/test/org.eclipse.smarthome.magic/src/main/java/org/eclipse/smarthome/magic/service/MagicMultiInstanceService.java

```diff
@@ -0,0 +1,51 @@
+/**
+ * Copyright (c) 2014,2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.smarthome.magic.service;
+
+import java.util.Map;
+import java.util.Map.Entry;
+
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.ConfigurationPolicy;
+import org.osgi.service.component.annotations.Modified;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Testing service for multi-context configurations.
+ *
+ * @author Stefan Triller - initial contribution
+ *
+ */
+@Component(configurationPolicy = ConfigurationPolicy.REQUIRE, service = MagicMultiInstanceService.class, configurationPid = "org.eclipse.smarthome.magicMultiInstance")
```

Comment:
this would be far more impressive if it had the `immediate = true` flag set, as nobody is using this service therefore it never gets activated. The `service = ...`  attribute however is not needed.

Reply:
If I omit the `service = ` attribute the services do not offer a service and they do not show up in paper UI...

I have set them to `immediate = true` though.